tests/conftest.py

```diff
@@ -19,10 +22,137 @@
 from rs_sqla_test_utils import db
 from rs_sqla_test_utils.utils import make_mock_engine
 
+logger = getLogger(__name__)
 
 _unicode = type(u'')
 
 
+@pytest.fixture(scope="session")
+def connection_kwargs(redshift_dialect_flavor):
+    """ Connection parameters for running integration tests
+    against an existing Redshift instance.
+
+    The tests are currently designed to work with the Travis CI
+    environment, where a Redshift instance is available
+    in the cloud. Travis CI passes the PGPASSWORD environment
+    variable and an API call to a Heroku app gets the rest
+    of the credentials.
+
+    This fixture allows the developer to pass in their own credentials
+    for other Redshift instances by setting the following environment
+    variables:
+
+        - REDSHIFT_HOST
+        - REDSHIFT_PORT
+        - REDSHIFT_USERNAME
+        - REDSHIFT_DATABASE
+        - PGPASSWORD
+
+    If these conditions are met, the tests will be ran
+    against a real Redshift instance. Otherwise, they
+    will be ran against a mock engine.
+
+    See the fixture, _redshift_database_tool, for usage.
+    """
+    pgpassword = os.environ.get('PGPASSWORD', None)
+    if not pgpassword:
+        pytest.skip('This test will only work on Travis.')
+
+    return {
+        "host": os.getenv("REDSHIFT_HOST", None),
+        "port": os.getenv("REDSHIFT_PORT", None),
+        "username": os.getenv("REDSHIFT_USERNAME", "travis"),
+        "password": pgpassword,
+        "database": os.getenv("REDSHIFT_DATABASE", "dev"),
+        "dialect": redshift_dialect_flavor,
+    }
+
+
+@pytest.fixture(scope="session")
+def aws_account_id():
+    """ The aws_account_id fixture allows the developer to pass in their own
+    AWS_ACCOUNT_ID for other Redshift instances by setting the following
+    environment variable: AWS_ACCOUNT_ID
+    """
+    return os.getenv("AWS_ACCOUNT_ID", '000123456789')
+
+
+@pytest.fixture(scope="session")
+def iam_role_name():
```

Comment:
for this logic around the role name & constructing the role arn from various pieces, could we rework this to use look for an env var e.g. `REDSHIFT_IAM_ROLE_ARN` that is the whole arn e.g. `arn:aws:iam::12345678912:role/ExampleRole`

Reply:
This was actually by choice. Some tests look for the ARN, and others look for the account_id and role_name. Passing in an ARN that doesn't reflect the account_id and role_name causes a few test failures due to the mismatch. 

The alternative would be to pass in the role arn and have the account_id, and role name fixtures parse that to get their values. If that's preferred, I can change my implementation.

Reply:
Sounds good, yes please update when you get a chance :)